### DISCUSSION TITLE
MQTT integration?

### DISCUSSION THREAD

**Discussion by switzer60** [PeteRager/lennoxs30#302] [Ideas]
Thanks so much for your work on making this HA integration work!  I've been battling with this dumb "smart" device for around 5 years now.  It looks like you have direct LAN connectivity, which is fantastic!  Based on your knowledge, what do you believe the level of effort would be to modify the codebase to integrate with MQTT instead of HA?  A direct integration with MQTT would open a -ton- of other options for home integration outside of just HA.  I'd love to hear your thoughts and estimation, and any willingness to consider it as a direction for this product?

Again, -thank you- for all the work that has gone into this integration!  It looks great (but I don't use HA!).

**Comment by PeteRager** [comment]
It'd be great to have an MQTT server.  It'd also be a lot of work as there is a lot of functionally. I would think a couple months of work could yield a limited beta.  Maybe it goes faster.  Certainly, having people helping would make it go quicker.

A lot of the code is separated out into a python API.  And the MQTT server could be built on that.  There is also logic in the integration (like the connection management) that should be moved to the API so the HA integration and MQTT server can use common code.


https://github.com/PeteRager/lennoxs30api
